index/scorch/merge.go

```diff
@@ -311,6 +314,10 @@ func (s *Scorch) mergeSegmentBases(snapshot *IndexSnapshot,
 		atomic.AddUint64(&s.stats.TotMemMergeErr, 1)
 		return nil, 0, err
 	}
+	err = zap.ValidateMerge(nil, sbs, sbsDrops, seg.(*zap.Segment))
+	if err != nil {
+		return nil, 0, fmt.Errorf("merge validation failed: %v", err)
```

Comment:
"merge validation failed:" -> looks exactly same across zap vs memory segments, Is it enough with the err to differentiate where it happened? meaning during zap or memory merging.. do we need to be slightly verbose to indicate the context as well